File: week6/sentimental-readability/readability.py
# import libraries
from cs50 import get_string


def main():
    # collect the text here
    text = get_string("Text: ")

    l, w, s = return_f(text)

    coleman = compute_coleman(l, w, s)

    # grade checking algth
    grade = round(coleman)
    if grade < 1:
        print("Before Grade 1")
    elif grade > 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")


def return_f(text):

    letter_count = 0
    word_count = 0
    sentence_count = 0

    # count number of letters in the text
    for i in text:
        if i.isalpha():
            letter_count += 1
    # print(letter_count)

    # count number of words in the text
    word = text.split()
    word_count = len(word)
    # print(word_count)

    # count number of sentences in the text
    for i in text:
        if i in [".", "!", "?"]:
            sentence_count += 1
    # print(sentence_count)
    return letter_count, word_count, sentence_count


def compute_coleman(l, w, s):
    # L average number of letters, the number of letters divided by the number of words, all multiplied by 100.
    l_average = (l / w) * 100
    # S average number of sentences, the number of sentences divided by the number of words, all multiplied by 100.
    s_average = (s / w) * 100
    coleman_index = (0.0588 * l_average) - (0.296 * s_average) - 15.8
    return coleman_index


if __name__ == "__main__":
    main()

File: week2/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
float compute_coleman(int letters, int words, int sentences);

int main(void)
{
    // Prompt the user for some text
    string text = get_string("Text: ");

    // Count the number of letters
    int letters = count_letters(text);

    //Count the number of words
    int words = count_words(text);

    // Count the number of sentences in the text
    int sentences = count_sentences(text);
    // printf("%i\n", sentences);

    // Compute the Coleman-Liau index
     float coleman_liau = compute_coleman(letters, words, sentences);

     // Print the grade level
     int grade = round(coleman_liau);
     if (grade < 1)
     {
        printf("Before Grade 1\n");
     }
     else if (grade > 16)
     {
        printf("Grade 16+\n");
     }
     else
     {
        printf("Grade %i\n", grade);
     }
}


int count_letters(string text)
{
    int count = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isalpha(text[i]))
        {
            count++;
        }
    }
    return count;
}

int count_words(string text)
{
    int count = 1;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == ' ')
        {
            count++;
        }
    }
    return count;
}

int count_sentences(string text)
{
    int count = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if ((text[i] == '.') || (text[i] == '!') || (text[i] == '?'))
        {
            count++;
        }
    }
    return count;
}

float compute_coleman(int letters, int words, int sentences)
{
    // L average number of letters, the number of letters divided by the number of words, all multiplied by 100.
    float l_average = ((float) letters / words) * 100.0;
    // S average number of sentences, the number of sentences divided by the number of words, all multiplied by 100.
    float s_average = ((float) sentences / words) * 100.0;
    float coleman_index = 0.0588 * l_average - 0.296 * s_average - 15.8;
    return coleman_index;
}

